### PR TITLE
Refactored Program.cs for best C# practices

### PR DESCRIPTION
**Pull request by reidbraswell1**
Ready for review.

### CODE REVIEW

File: RefactorKata/Program.cs [hunk 1]
```diff
@@ -7,35 +7,32 @@ class Program
     {
         static void Main(string[] args)
         {
-            //This is intentionally bad : (  Let's Refactor!
-            System.Data.SqlClient.SqlConnection Conn = new System.Data.SqlClient.SqlConnection("Server=.;Database=myDataBase;User Id=myUsername;Password = myPassword;");
+            const string SQL_COMMAND_TEXT = "select * from Products";
```
Comment (codytwinton) [inline RefactorKata/Program.cs]: C# naming convention for `const` is still PascalCase. Change `SQL_COMMAND_TEXT` to `SqlCommandText`.

File: RefactorKata/Program.cs [hunk 2]
```diff
@@ -7,35 +7,32 @@ class Program
...
+            const string SERVER_INFO = "Server=.;Database=myDataBase;User Id=myUsername;Password = myPassword;";
```
Comment (codytwinton) [inline RefactorKata/Program.cs]: C# naming convention for `const` is still PascalCase. Also, it's referred to as a connection string. I'd change `SERVER_INFO` to `ConnString` or `ConnectionString`.

File: RefactorKata/Program.cs [hunk 3]
```diff
@@ -7,35 +7,32 @@ class Program
...
-            cmd.CommandText = "select * from Products";
-            /*
-             * cmd.CommandText = "Select * from Invoices";
-             */
-            System.Data.SqlClient.SqlDataReader reader = cmd.ExecuteReader();
             List<Product> products = new List<Product>();
 
-            //TODO: Replace with Dapper
-            while (reader.Read())
+            using (var conn = new System.Data.SqlClient.SqlConnection(SERVER_INFO))
```
Comment (codytwinton) [inline RefactorKata/Program.cs]: Add a using directive at the very top of your file of `using System.Data.SqlClient;`. Then, you can just use `SqlConnection(ConnString)`

File: RefactorKata/Program.cs [hunk 4]
```diff
@@ -7,35 +7,32 @@ class Program
...
-                products.Add(prod);
-            }
-            Conn.Dispose();
+                var cmd = conn.CreateCommand();
+                cmd.CommandText = SQL_COMMAND_TEXT;   
+                var reader = cmd.ExecuteReader();
+                while(reader.Read())
+                {
+                    var prod = new Product();
+                    prod.Name = reader["Name"].ToString();
```
Comment (codytwinton) [inline RefactorKata/Program.cs]: I'd like to see the object initializer here.

File: RefactorKata/Program.cs [hunk 5]
```diff
@@ -7,35 +7,32 @@ class Program
...
+                    products.Add(prod);
+                }//while//
+            }//using//
             Console.WriteLine("Products Loaded!");
             for (int i =0; i< products.Count; i++)
```
Comment (codytwinton) [inline RefactorKata/Program.cs]: This should be a `foreach` loop.

File: RefactorKata/Program.cs [hunk 6]
```diff
@@ -7,35 +7,32 @@ class Program
...
             {
-                Console.WriteLine(products[i].name);
-            }
-        }
-    }
+                Console.WriteLine(products[i].Name);
+            }//for//
+        }//Main//
+    }//Program//
     public class Product
```
Comment (codytwinton) [inline RefactorKata/Program.cs]: Move the Product class to its own file.

File: RefactorKata/Program.cs [hunk 7]
```diff
@@ -7,35 +7,32 @@ class Program
...
     {
-        public string name;
-        public string Name { get { return name; } set { name = value; } }
-    }
-}
+        public string Name { get;   set; }
```
Comment (codytwinton) [inline RefactorKata/Program.cs]: Only 1 space is needed between `get;` and `set;`, but I see 3.

**Comment by reidbraswell1** [comment]
Ready for re-review.

File: RefactorKata/Program.cs [hunk 8]
```diff
@@ -1,41 +1,37 @@
...
+            const string SqlCommandText = "select * from Products";
+            const string ConnectionString = "Server=.;Database=myDataBase;User Id=myUsername;Password = myPassword;";
 
-            System.Data.SqlClient.SqlCommand cmd = Conn.CreateCommand();
-            cmd.CommandText = "select * from Products";
-            /*
-             * cmd.CommandText = "Select * from Invoices";
-             */
-            System.Data.SqlClient.SqlDataReader reader = cmd.ExecuteReader();
             List<Product> products = new List<Product>();
```
Comment (codytwinton) [inline RefactorKata/Program.cs]: Don't forget the `var` keyword here.

File: RefactorKata/Program.cs [hunk 9]
```diff
@@ -1,41 +1,37 @@
...
-                products.Add(prod);
-            }
-            Conn.Dispose();
+                var cmd = conn.CreateCommand();
+                cmd.CommandText = SqlCommandText;
+                var reader = cmd.ExecuteReader();
+                while (reader.Read())
+                {
+                    var prod = new Product()
+                    {
```
Comment (codytwinton) [inline RefactorKata/Program.cs]: You can probably single line this statement: `var prod = new Product { Name = reader["Name"].ToString() };`

**Comment by reidbraswell1** [comment]
Made requested changes ready for review again.

File: RefactorKata/Product.cs [hunk 10]
```diff
@@ -0,0 +1,4 @@
+public class Product
+{
+    public string Name { get; set; }
+}//Product//
```
Comment (codytwinton) [inline RefactorKata/Product.cs]: I'd remove the comments and add a single newline at the end of each file.

File: RefactorKata/Program.cs [hunk 11]
```diff
@@ -1,41 +1,34 @@
...
             {
-                var prod = new Product();
-                prod.name = reader["Name"].ToString();
-                products.Add(prod);
+                var cmd = conn.CreateCommand();
+                cmd.CommandText = SqlCommandText;
+                var reader = cmd.ExecuteReader();
+                while (reader.Read())
+                {
+                    var prod = new Product() { Name = reader["Name"].ToString() };
```
Comment (codytwinton) [inline RefactorKata/Program.cs]: Parenthesis `()` can be inferred/removed when using the object initializer.

**Pull request merged**